setuptools/msvc.py

```diff
@@ -751,16 +755,16 @@ def NetFxSdkVersion(self):
         )
 
     @property
-    def NetFxSdkDir(self):
+    def NetFxSdkDir(self) -> str | None:
         """
         Microsoft .NET Framework SDK directory.
 
         Return
         ------
-        str
+        str | None
             path
         """
-        sdkdir = ''
+        sdkdir: str | None = ''
         for ver in self.NetFxSdkVersion:
             loc = os.path.join(self.ri.netfx_sdk, ver)
             sdkdir = self.ri.lookup(loc, 'kitsinstallationfolder')
```

Comment:
Because `RegistryInfo.lookup` can return `None`, and this method returns the last falsy result if all falsy. Then the return type is potentially `None`.